floop.go

```diff
@@ -49,9 +52,23 @@ func New(conf *Config, input *child.NewInput) (*Floop, error) {
 
 	flp.procInput = input
 	flp.proc, err = child.New(flp.procInput)
+	go flp.listenSignals(flp.proc.Signal)
+
 	return flp, err
 }
 
+func (floop *Floop) listenSignals(sigProcesser func(os.Signal) error) {
+	signalChannel := make(chan os.Signal, 3)
```

Comment:
You shouldn't need a buffer size larger than 1 for the signal channel. Once a single signal is sent to the channel, any further signals sent should block, as you only want/need to process the first signal.

Reply:
oh, right